src/kernel/validation_notifications_interface.h

```diff
@@ -27,6 +28,7 @@ class ValidationNotifications
     virtual void notifyHeaderTip(SynchronizationState state, int64_t height, int64_t timestamp, bool presync) = 0;
     virtual void showProgress(const std::string& title, int nProgress, bool resume_possible) = 0;
     virtual void doWarning(const bilingual_str& warning) = 0;
+    virtual bool fatalError(const std::string& strMessage, bilingual_str user_message = {}) = 0;
```

Comment:
In commit "refactor: Pass FatalError in options to ChainstateManager" (2f9c2d245360b3fad6d469a76c2916d75b027b57)

I think this signature should be cleaned up and not be added to kernel like this. Would suggest:

- Changing `bool` return type to `void`. The return value is never used and the only value that is ever returned is false. Having the return value makes the interface confusing because it's unclear how the value even could be used.
- Renaming `strMessage` parameter to `debug_message` to modernize the name and distinguish the technical error message which is meant to go to the debug log from the translated user message.